### ISSUE TITLE
Disable timeout during a Sapper export

### ISSUE THREAD

**Issue by Flameborn** [nolanlawson/pinafore#1924]
When building Pinafore on a Raspberry Pi 3, the export server does not start up quick enough.

By default, Sapper defines a 5000 ms timeout when exporting, but this seems to be too short and the build fails with a timeout when crawling http://localhost:22939.

Sapper has a *--no-timeout* flag to disable this, which should allow successful builds on slower machines as well. It would be nice if this was added to the build steps.

**Comment by Flameborn** [comment]
Unfortunately, the timeout does not seem to be honored, I still get an error while building via *yarn build*:

```
> Crawling http://localhost:22939/
> timed out waiting for connection
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "sapper-export" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "build-steps" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by nolanlawson** [comment]
@Flameborn Did you re-run `yarn` ?

**Comment by nolanlawson** [comment]
BTW a temporary workaround is to build the project on a machine other than the Raspberry Pi, then copy all the files over to the Pi and run the server.

**Comment by nolanlawson** [comment]
FWIW I tried building commit 1e97482 on my Raspberry Pi 4 and I could run `yarn && yarn build && yarn start` successfully.

**Comment by Flameborn** [comment]
>  Did you re-run    yarn    ?

Yes. I ran *yarn --production --pure-lockfile*, then *yarn build*.

> BTW a temporary workaround is to build the project on a machine other than the Raspberry Pi, then copy all the files over to the Pi and run the server.

This is what I am doing for now, I just copy over the \_\_sapper\_\_ directory after building. Thank you for the hint!

> FWIW I tried building commit 1e97482 on my Raspberry Pi 4 and I could run yarn && yarn build && yarn start successfully.

I'm guessing that app startup on the Pi 3B is not as quick in this case as it's on the Pi 4. It still seems like the timeout is 5 seconds here.

I'm getting the same timeout via ba3b76f7

**Comment by nolanlawson** [comment]
OK, reopening the issue. If you can do any debugging to figure out why the 5 minute (not 5 second) timeout is applying, let me know. I can't seem to reproduce on my Pi 4.

**Comment by Flameborn** [comment]
I'm unfortunately not familiar with Sapper, but I'll see if I can figure out something.

Btw, is this relevant? https://github.com/sveltejs/sapper/issues/890

**Comment by nolanlawson** [comment]
This is probably related, yeah. I have no idea why it's timing out.

**Comment by Flameborn** [comment]
An update on this issue, with the latest commit at the time of this comment, via cbdbb059262f81a5c0e7683dcf8ac8737db650cb, building under MacOS Big Sur 11.4 also times out when crawling the sapper port, when using Node 16.4.0.

**Comment by nolanlawson** [comment]
I wonder what conditions cause the timeout. I use Vercel to build and publish Pinafore, and I've never seen this issue happen on the Vercel servers. Nor does it happen in the Circle CI tests that I'm aware of.

**Comment by Fastidious** [comment]
I can't build on my Linux VPS either, same errors:
```
> Building...
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "sapper-export" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "build-steps" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
The `yarn build` step takes my 2GB VPS to an unworkable state too. It overloads it really badly.
